[Docs] Add docs/reference split (#795)
exercism/v3: .github/ISSUE_TEMPLATE/new-reference-doc.md

---

name: "[MAINTAINERS] Add new reference document"
about:
FOR MAINTAINERS ONLY - Use this template to create an issue to add a new reference
document
title: "[Docs] Add new reference document: <DOC>"
labels: status/help-wanted, type/new-reference-doc
assignees: ""

> ---[Use this issue template to describe how to add a new reference doc. Only suggest adding a new reference doc when no existing reference document has been written. Please fill in the issue template, and remove any sections wrapped in square brackets (such as this section!).]

Reviewer: ```suggestion
---

[Use this issue template to describe how to add a new reference doc. Only suggest adding a new reference doc when no existing reference document has been written. Please fill in the issue template, and remove any sections wrapped in square brackets (such as this section!).]
```

Not sure if Markdown frontmatter will work if there's text in the same line as the closing `---`.

This issue describes how to add a new reference document: <DOC>.

## Description

> [A clear description of what information the reference document should contain. It should be clear how this document can help a student.

Reviewer 2: ```suggestion
[A clear description of what information the reference document should contain. This document should help other maintainers developing and building context exercises
```

Maybe?

> [A clear description of what information the reference document should contain. This document should help other maintainers developing and building context exercises

Reviewer 2: ```suggestion
[A clear description of what information the reference document should contain. This document should help other maintainers developing and building context exercises.
```

Sorry :)

For example:

Concept `X` is used to do `Y`]

### Contributing

[Describe the exact steps the contributor has to take to add the document.

For example:

- Create the document in the appropriate location in the `reference` folder.

> - Check to see if there are other document that can refer to the new document.

Reviewer 3: ```suggestion
- Check to see if there are other documents that can refer to the new document.
```
